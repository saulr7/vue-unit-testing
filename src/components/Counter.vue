<template>
  <div style="display: inline;">
    <h2>{{ getTittle }}</h2>
    <button @click="sub">-</button>
    <h3>value: {{ this.value }}</h3>
    <button @click="add">+</button>
  <p>{{ value }} <sup>2</sup> = {{ squareCounter }}</p>
  <p data-testid="counter">{{ value }} </p>
  </div>

</template>

<script>
export default {
  name: "Counter",
  props: {
    initValue: {type :Number, required : true, default : 100},
    title: String,
  },
  data() {
    return { value: this.initValue };
  },
  methods: {
    square() {
      console.log("sqaure");
      return this.value * this.value;
    },
    sub() {
      this.value--;
    },
    add() {
      this.value++;
    },
  },
  computed: {
    squareCounter() {
      return this.value * this.value;
    },
    getTittle(){
      return this.title || "counter"
    }
  },
};
</script>

<style></style>
