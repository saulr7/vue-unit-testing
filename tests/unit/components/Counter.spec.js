import { shallowMount } from "@vue/test-utils";
import Counter from "@/components/Counter.vue";

describe("Counter Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Counter);
  });

  test("should match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should have default value", () => {
    const h2 = wrapper.find("h2");
    expect(h2.text()).toBe("counter");
  });

  test("should have default value === 100", () => {
    // const pTgas = wrapper.findAll("p");
    // expect(pTgas[1].text()).toBe("100");
    const pTga = wrapper.find("[data-testid='counter'");
    expect(pTga.text()).toBe("100");
  });

  test("should increase value", async () => {
    const btnincrease = wrapper.findAll("button").at(1);
    await btnincrease.trigger("click");
    const pTga = wrapper.find("[data-testid='counter'");
    expect(pTga.text()).toBe("101");
  });

  test("should increase value", async () => {
    const btnDecrease = wrapper.findAll("button").at(0);
    await btnDecrease.trigger("click");
    await btnDecrease.trigger("click");
    const pTga = wrapper.find("[data-testid='counter'");
    expect(pTga.text()).toBe("98");
  });

  test("should use default value", () => {
    const { initValue } = wrapper.props();
    expect(initValue).toBe(100);
  });

  test("should use props", () => {
    const title = "hey";
    const wrapper = shallowMount(Counter, {
      props: {
        title,
      },
    });

    expect(wrapper.find("h2").text()).toBe(title);
  });
});
