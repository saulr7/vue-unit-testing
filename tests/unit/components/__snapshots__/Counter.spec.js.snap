// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component should match snapshot 1`] = `
<div style="display: inline;">
  <h2>counter</h2><button>-</button>
  <h3>value: 100</h3><button>+</button>
  <p>100 <sup>2</sup> = 10000</p>
  <p data-testid="counter">100</p>
</div>
`;
